gems/aws-sdk-core/lib/aws-sdk-core/rest/request/body.rb

```diff
@@ -17,7 +17,12 @@ def initialize(serializer_class, rules)
         # @param [Seahorse::Client::Http::Request] http_req
         # @param [Hash] params
         def apply(http_req, params)
-          http_req.body = build_body(params)
+          body = build_body(params)
+          # for rest-json, ensure we send at least an empty object for
+          if body.nil? && @serializer_class == Json::Builder && @rules[:payload]
```

Comment:
I think the `payload` part here is wrong. But I think this is the correct place to put the hack.. This class is shared so we need to check the serializer class.